pkg/registry/apis/dashboard/legacysearcher/search_client.go

```diff
@@ -28,29 +32,92 @@ func (c *DashboardSearchClient) Search(ctx context.Context, req *resource.Resour
 		return nil, err
 	}
 
-	if req.Query == "*" {
-		req.Query = ""
+	if strings.Contains(req.Query, "*") {
+		req.Query = strings.ReplaceAll(req.Query, "*", "")
 	}
 
 	// TODO add missing support for the following query params:
-	// - tag
-	// - starred (won't support)
-	// - page (check)
-	// - type
-	// - sort
-	// - deleted
+	// - folderIds (won't support, must use folderUIDs)
 	// - permission
-	// - dashboardIds
-	// - dashboardUIDs
-	// - folderIds
-	// - folderUIDs
-	// - sort (default by title)
 	query := &dashboards.FindPersistedDashboardsQuery{
 		Title: req.Query,
 		Limit: req.Limit,
+		Page:  req.Page,
 		// FolderUIDs:   req.FolderUIDs,
-		Type:         searchstore.TypeDashboard,
 		SignedInUser: user,
+		IsDeleted:    req.IsDeleted,
+	}
+
+	var queryType string
+	if req.Options.Key.Resource == "dashboards" {
```

Comment:
we hardcode "dashboards" and "folders" quite a bit.  a couple constants would help ( surprised these don't exist already )

Reply:
good idea - you can use `dashboard.DASHBOARD_RESOURCE` from pkg/apis/dashboard